crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -298,6 +303,11 @@ where
         CacheTaskHandle { cache, to_prewarm_task: Some(to_prewarm_task), cache_metrics }
     }
 
+    /// Takes the trie input from the inner payload processor, if it exists.
+    pub fn take_trie_input(&mut self) -> Option<TrieInput> {
+        self.trie_input.take().map(|input| input.cleared())
+    }
```

Comment:
I see, this is the only way how we can obtain the trieinput again and we ensure here that it's cleared